test/FlashLoan.t.sol

```diff
@@ -0,0 +1,279 @@
+        vm.startPrank(userEOA);
+        vm.expectEmit(true, true, true, true);
+        emit FastLaneErrorsEvents.SolverTxResult(address(solver), solverOneEOA, true, false, 1048578);
+        vm.expectRevert();
+        atlas.metacall({ userOp: userOp, solverOps: solverOps, dAppOp: dAppOp });
+        vm.stopPrank();
+
+        // now try it again with a valid solverOp - but dont fully pay back
+        solverOps[0] = txBuilder.buildSolverOperation({
+            userOp: userOp,
+            solverOpData: abi.encodeWithSelector(SimpleSolver.onlyPayBid.selector, 1e18),
+            solverEOA: solverOneEOA,
+            solverContract: address(solver),
+            bidAmount: 1e18,
+            value: 10e18
+        });
+
+        (sig.v, sig.r, sig.s) = vm.sign(solverOnePK, atlasVerification.getSolverPayload(solverOps[0]));
+        solverOps[0].signature = abi.encodePacked(sig.r, sig.s, sig.v);
+
+        dAppOp = txBuilder.buildDAppOperation(governanceEOA, userOp, solverOps);
+        (sig.v, sig.r, sig.s) = vm.sign(governancePK, atlasVerification.getDAppOperationPayload(dAppOp));
+        dAppOp.signature = abi.encodePacked(sig.r, sig.s, sig.v);
+
+        // Call again with partial payback, should still revert
+        vm.startPrank(userEOA);
+        vm.expectEmit(true, true, true, true);
+        emit FastLaneErrorsEvents.SolverTxResult(address(solver), solverOneEOA, true, false, 8388610);
+        vm.expectRevert();
+        atlas.metacall({ userOp: userOp, solverOps: solverOps, dAppOp: dAppOp });
+        vm.stopPrank();
+
+        // final try, should be successful with full payback
+        solverOps[0] = txBuilder.buildSolverOperation({
+            userOp: userOp,
+            solverOpData: abi.encodeWithSelector(SimpleSolver.payback.selector, 1e18),
+            solverEOA: solverOneEOA,
+            solverContract: address(solver),
+            bidAmount: 1e18,
+            value: 10e18
+        });
+
+        (sig.v, sig.r, sig.s) = vm.sign(solverOnePK, atlasVerification.getSolverPayload(solverOps[0]));
+        solverOps[0].signature = abi.encodePacked(sig.r, sig.s, sig.v);
+
+        dAppOp = txBuilder.buildDAppOperation(governanceEOA, userOp, solverOps);
+        (sig.v, sig.r, sig.s) = vm.sign(governancePK, atlasVerification.getDAppOperationPayload(dAppOp));
+        dAppOp.signature = abi.encodePacked(sig.r, sig.s, sig.v);
+
+        uint256 solverStartingWETH = WETH.balanceOf(address(solver));
+        uint256 atlasStartingETH = address(atlas).balance;
+        uint256 userStartingETH = address(userEOA).balance;
+
+        assertEq(solverStartingWETH, 1e18, "solver incorrect starting WETH");
+        assertEq(atlasStartingETH, 102e18, "atlas incorrect starting ETH"); // 2e initial + 1e solver + 100e user deposit
+
+        // Last call - should succeed
+        vm.startPrank(userEOA);
+        atlas.metacall({ userOp: userOp, solverOps: solverOps, dAppOp: dAppOp });
+        vm.stopPrank();
```

Comment:
Why no `expectEmit` on the successful call?
```solidity
uint256 result = (1 << uint256(SolverOutcome.Success)) | (1 << uint256(SolverOutcome.ExecutionCompleted));
vm.expectEmit(true, true, true, true);
emit FastLaneErrorsEvents.SolverTxResult(address(solver), solverOneEOA, true, true, result);
```

Reply:
It reverts on unfulfilled so I had just left it like that originally but add that in